### ISSUE TITLE
"Location Should Not Be Hardcoded" false positive when location parameter isn't named "location"

### ISSUE THREAD

**Issue by aholler2** [Azure/arm-ttk#454]
The parameter for location in my main template is not named "location", but "storagelocation". This fails under the test logic here:
```
if ($TemplateWithoutLocationParameter -like '*resourceGroup().location*' # -or
    # $TemplateWithoutLocationParameter -like '*deployment().location*'
) {
    # If it did, write an error
    Write-Error "$TemplateFileName must use the location parameter, not resourceGroup().location or deployment().location (except when used as a default value in the main template)" -ErrorId Location.Parameter.Should.Be.Used -TargetObject $parameter
}
```

Sample ARM template:
```
{
    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
    "contentVersion": "#{versionNumber}#",
    "metadata": {
        "title": "Storage Account",
        "description": "This template provisions a Storage Account"
      },
    "parameters": {
        "storageAccountName": {
            "type": "string",
            "metadata": {
                "description": "(Required) The name of the storage account. Storage account names must be 3-24 characters in length and only use numbers and lower-case letters."
            }
        },
        "accountType": {
            "type": "string",
            "defaultValue": "Standard_LRS",
            "metadata": {
                "description": "Replication option SKU. Defaults to Standard Locally Redundant Storage."
            }
        },
        "accessTier": {
            "type": "string",
            "defaultValue": "Hot",
            "metadata": {
                "description": "Tier indicating pattern of data usage. Defaults to Hot for frequent data access."
            }
        },
        "storageTagValues": {
            "type": "object",
            "defaultValue": {
            },
            "metadata": {
                "description": "Tag name/value pairs defined on the storage account resource."
            }
        },
        "storagelocation": {
            "type": "string",
            "defaultValue": "[resourceGroup().location]",
            "metadata": {
                "description": "Azure region to deploy the storage account to. Defaults to resource group location."
            }
        },
        "kind": {
            "type": "string",
            "defaultValue": "StorageV2",
            "metadata": {
                "description": "Indicates the type of storage account. Defaults to StorageV2 supporting blobs, files, queues, and tables."
            }
        },
        "advancedThreatProtectionEnabled": {
            "type": "bool",
            "defaultValue": false,
            "metadata": {
                "description": "Enable or disable Advanced Threat Protection on Blob service. Defaults to false."
            }
        },
        "largeFileSharesEnabled": {
            "type": "bool",
            "defaultValue": false,
            "metadata": {
                "description": "Increase max file share size from 5 TiB to 100 TiB. Cannot convert account to geo-redundant replication and cannot be disabled/reversed. Defaults to false."
            }
        },
        "minimumTlsVersion": {
            "type": "string",
            "defaultValue": "TLS1_2",
            "metadata": {
                "description": "Set the minimum TLS version to be permitted on requests to storage. Defaults to TLS1_2."
            }
        }
    },
    "variables": {},
    "resources": [
        {
            "apiVersion": "2019-06-01",
            "name": "[parameters('storageAccountName')]",
            "location": "[parameters('storageLocation')]",
            "type": "Microsoft.Storage/storageAccounts",
            "tags": "[parameters('storageTagValues')]",
            "identity": {
                "type": "SystemAssigned"
            },
            "sku": {
                "name": "[parameters('accountType')]"
            },
            "kind": "[parameters('kind')]",
            "properties": {
                "accessTier": "[if(equals(parameters('kind'),'Storage'), json('null'), parameters('accessTier'))]",
                "largeFileSharesState": "[if(equals(parameters('largeFileSharesEnabled'), bool('false')), json('null'), 'Enabled')]",
                "encryption": {
                    "services": {
                        "blob": {
                            "enabled": true
                        },
                        "file": {
                            "enabled": true
                        },
                        "table": {
                            "enabled": true
                        },
                        "queue": {
                            "enabled": true
                        }
                    },
                    "keySource": "Microsoft.Storage"
                },
                "supportsHttpsTrafficOnly": true,
                "minimumTlsVersion": "[parameters('minimumTlsVersion')]"
            },
            "resources": [
                {
                    "condition": "[parameters('advancedThreatProtectionEnabled')]",
                    "type": "providers/advancedThreatProtectionSettings",
                    "name": "Microsoft.Security/current",
                    "apiVersion": "2017-08-01-preview",
                    "dependsOn": [
                        "[resourceId('Microsoft.Storage/storageAccounts/', parameters('storageAccountName'))]"
                    ],
                    "properties": {
                        "isEnabled": true
                    }
                }
            ]
        }
    ],
    "outputs": {
        "storageAccountName": {
            "type": "string",
            "value": "[parameters('storageAccountName')]"
        },
        "storageAccountResourceId": {
            "type": "string",
            "value": "[resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]"
        }
    }
}
```

**Comment by bmoore-msft** [comment]
The test currently removes the location parameter from the template and then does a textual search... I think in order for this to be fixed, we'd have to assert that *any* parameter can use [resourceGroup().location] as a defaultValue.  That seems to be ok as the intent is to allow the user to override the location properties with a different value and if it's a param should work.

Would be worth getting some other opinions on this though...

**Comment by bmoore-msft** [comment]
s/b fixed in #577
